xbmc/favourites/GUIWindowFavourites.cpp

```diff
@@ -128,19 +148,21 @@ bool CGUIWindowFavourites::OnSelect(int item)
   if (!favURL.IsValid())
     return false;
 
-  if (favURL.GetAction() == CFavouritesURL::Action::PLAY_MEDIA)
+  CFileItem targetItem{favURL.GetTarget(), favURL.IsDir()};
+  targetItem.LoadDetails();
+
+  const bool isNoFolderOrIsPlayMedia{!targetItem.m_bIsFolder ||
```

Comment:
I think it reads better if you decouple this into the actual set of conditions in the if statement. I mean `isNoFolderOrIsPlayMedia` doesn't really group together anything meaningful (it's redundant according to to the actual conditions)

Reply:
Sure, I can do this.